hoodie-common/src/main/java/com/uber/hoodie/common/util/FSUtils.java

```diff
@@ -201,31 +203,53 @@ public static String getRelativePartitionPath(Path basePath, Path partitionPath)
         partitions.add(getRelativePartitionPath(basePath, filePath.getParent()));
       }
       return true;
-    });
+    }, true);
     return partitions;
   }
 
   public static final List<String> getAllDataFilesForMarkers(FileSystem fs, String basePath, String instantTs,
       String markerDir) throws IOException {
     List<String> dataFiles = new LinkedList<>();
-    FSUtils.processFiles(fs, markerDir, (status) -> {
+    processFiles(fs, markerDir, (status) -> {
```

Comment:
Would nt a better strategy be to just list the `.hoodie` folder for marker files? I feel we are listing the entire table and then throwing it away?

Reply:
your root is different.. got it